### ISSUE TITLE
Noise on rerun of existing Web App Sites

### ISSUE THREAD

**Issue by martyh888** [Azure/arm-template-whatif#204]
## Describe the noise
I am using Bicep and deploying with what-if and when I try to use sites Ia m getting noise on location (this happens on the second run to replace or update so when there is no change I still get that the location needs to be deleted in most objects under the root sites (sites/xxx/networkConfig/virtualNetwork, sites/xxx/hostNameBindings/bindingname, sites/xxx/siteextensions)

**Resource type** (i.e. Microsoft.Storage/storageAccounts)
Microsoft.Web/sites
Microsoft.Web/sites/siteextensions@2020-06-01

**apiVersion** (i.e. 2019-04-01)

**Client (PowerShell, Azure CLI, or API)**
AZ.Resource 4.2
Az.Azure 6.1

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**

The API doesn't change anything seems to be an issue on all but I am using @2020-06-01 generally

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**

I expect no noise as there was no change and i passed the values on add before

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**

~ Microsoft.Web/sites/xxx [2020-06-01]
    + properties.siteConfig.localMySqlEnabled:   false
    + properties.siteConfig.netFrameworkVersion: "v4.0"

  ~ Microsoft.Web/sites/xxx/hostNameBindings/xxx.azurewebsites.us [2020-06-01]
    - location: "USGov Texas"

  ~ Microsoft.Web/sites/xxx/networkConfig/virtualNetwork [2020-06-01]
    - location: "USGov Texas"

  ~ Microsoft.Web/sites/xxx/siteextensions/AspNetCoreRuntime.3.1.x64 [2020-06-01]
    - location: "USGov Texas"
    - properties:

        id:   "AspNetCoreRuntime.3.1.x64"
        type: "Gallery"

**Additional context**

**Comment by torreymicrosoft** [comment]
Thanks for your patience, and sorry for the long wait on this one. It was recently picked up as one of the older open issues for the ARM Deployments team to review.

Going back through your output, the noise you captured comes from two causes, and both are already tracked here with fuller diagnosis attached:

- `+ properties.siteConfig.localMySqlEnabled` and `+ properties.siteConfig.netFrameworkVersion` on the site itself is the `Microsoft.Web/sites` siteConfig behaviour tracked in #90. Alex set out the root cause there in August 2021: the Web RP only returns the full `siteConfig` payload behind a query string on GET, what-if has no way to add that query string today, so it compares against a reduced response and reports the gap. Someone from the Web side confirmed the same limitation on that thread the following day.
- `- location` on `hostNameBindings`, `networkConfig` and `siteextensions` is the child resource `location` noise tracked in #171, which has the same signature on `Microsoft.Web/sites/hostNameBindings`.

Since this issue does not add a third cause beyond those two, I am going to consolidate rather than keep three threads running on the same behaviour.

Duplicate of #90 for the siteConfig half, with the child resource `location` half tracked in #171.

If you are seeing Web app noise that neither of those covers, please reopen or open a fresh issue with the what-if output and the API version, and I will pick it up. Worth flagging your output is from Azure Government, so if the behaviour there ever diverges from public cloud, that is genuinely useful to know and worth its own issue.
